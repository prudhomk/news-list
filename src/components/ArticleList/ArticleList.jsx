import React from 'react';
import PropTypes from 'prop-types';
import Article from '../Article/Article';

const ArticleList = ({ articles }) => {
  if(articles) {
    const articleDetails = articles.map((article, i) => (
      <>
        <li key={Date.now() + i}>
          <Article {...article}/>
        </li>
      </>
    ));
  
    return <ul>{articleDetails}</ul>;
  }
  return <ul>AAAAAAAAHHHHHHHHH</ul>;
}; 
  

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      description: PropTypes.string,
    })
  )
};

export default ArticleList;
