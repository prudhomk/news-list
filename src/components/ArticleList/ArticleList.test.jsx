import React from 'react';
import { cleanup, render } from '@testing-library/react';
import ArticleList from '../ArticleList/ArticleList';

describe('ArticleList component', () => {
  afterEach(() => cleanup());
  it('renders articlelist', async () => {
    const { asFragment } = render(<ArticleList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
