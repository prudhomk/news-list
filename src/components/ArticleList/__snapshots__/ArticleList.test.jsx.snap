// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component renders articlelist 1`] = `
<DocumentFragment>
  <ul>
    AAAAAAAAHHHHHHHHH
  </ul>
</DocumentFragment>
`;
